src/tasks/command/command_parser.c

```diff
@@ -0,0 +1,56 @@
+/**
+ * @author  Thomas Haile
+ * @date    2025-05-24
+ *
+ * Command parsing implementation
+ */
+
+#include "command_parser.h"
+#include "macros.h"
+
+/// @brief Parse packet and dispatch command to appropriate queue
+void dispatch_command(slate_t *slate, packet_t *packet)
+{
+    slate->number_commands_processed++;
+
+    Command command_id = (Command)packet->data[0];
+    LOG_INFO("Command ID Received: %i", command_id);
+
+    switch (command_id)
+    {
+        /* Payload Commands */
+        case TAKE_PHOTO:
+        case DOWNLOAD_PHOTO:
+        case TAKE_AND_SEND_PHOTO:
+        {
+            PAYLOAD_COMMAND_DATA task;
+            strncpy(task.serialized_command, packet->data + 1,
+                    sizeof(task.serialized_command) - 1);
+            task.serialized_command[sizeof(task.serialized_command) - 1] = '\0';

```

Comment:
This always ensures that the final bytes of the copied-to string field is a null-terminator. **This feels more widely useful.** We should abstract it into some util function that does a 'safe-copy' of strings (in this way).